src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -356,12 +360,9 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
                     if (_options.Adapters.Any(adapter => adapter.NeedsRefresh()) || changedKeyValuesCollection?.Any() == true || keyValueChanges.Any())
                     {
                         _serverData = serverData;
+                        _watchedSettings = watchedSettings;
 
-                        if (refreshAll)
-                        {
-                            UpdateWatchedSettings();
-                        } 
-                        else
+                        if (!refreshAll)

```

Comment:
Can you revert the changes to this code block since the underlying logic has not changed here. 
Just want to minimize the code we touch - less chance of introducing new bugs :)